tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp

```diff
@@ -0,0 +1,213 @@
+    }
+};
+
+template <typename CharT, typename callback_type>
+constexpr void test_parse_helper(const CharT* (*func)(const CharT*, const CharT*, callback_type&&),
+    basic_string_view<CharT> view, bool err_expected = false,
+    typename basic_string_view<CharT>::size_type expected_end_position = basic_string_view<CharT>::npos,
+    callback_type&& callbacks                                          = {}) {
+    try {
+        auto end = func(view.data(), view.data() + view.size(), move(callbacks));
+        if (expected_end_position != basic_string_view<CharT>::npos) {
+            assert(end == view.data() + expected_end_position);
+        }
+        assert(!err_expected);
+    } catch (const format_error&) {
+        assert(err_expected);
+    }
+}
+
+template <typename CharT>
+constexpr bool test_parse_conversion_spec() {
+    auto parse_conv_spec_fn = _Parse_conversion_specs<CharT, testing_callbacks<CharT>>;
+    using view_typ          = basic_string_view<CharT>;
+
+    view_typ s0(TYPED_LITERAL(CharT, "%B"));
+    view_typ s1(TYPED_LITERAL(CharT, "%Ec"));
+    view_typ s2(TYPED_LITERAL(CharT, "%Od"));
+    view_typ s3(TYPED_LITERAL(CharT, "%E"));
+
+    vector<_Chrono_specs<CharT>> v0{_Chrono_specs<CharT>{'\0', 'B'}};
+    test_parse_helper(parse_conv_spec_fn, s0, false, view_typ::npos, {.expected_chrono_specs = v0});
+
+    vector<_Chrono_specs<CharT>> v1{_Chrono_specs<CharT>{'E', 'c'}};
+    test_parse_helper(parse_conv_spec_fn, s1, false, view_typ::npos, {.expected_chrono_specs = v1});
+
+    vector<_Chrono_specs<CharT>> v2{_Chrono_specs<CharT>{'O', 'd'}};
+    test_parse_helper(parse_conv_spec_fn, s2, false, view_typ::npos, {.expected_chrono_specs = v2});
+
+    if (!is_constant_evaluated()) {
+        vector<_Chrono_specs<CharT>> v{};
+        test_parse_helper(parse_conv_spec_fn, s3, true, view_typ::npos, {.expected_chrono_specs = v});
+    }
+
+    return true;
+}
+
+template <typename CharT>
+constexpr bool test_parse_chrono_format_specs() {
+    auto parse_chrono_format_specs = _Parse_chrono_format_specs<CharT, testing_callbacks<CharT>>;
+    using view_typ                 = basic_string_view<CharT>;
+
+    view_typ s0(TYPED_LITERAL(CharT, "%Oe"));
+    view_typ s1(TYPED_LITERAL(CharT, "lit"));
+    view_typ s2(TYPED_LITERAL(CharT, "%H:%M}"));
+    view_typ s3(TYPED_LITERAL(CharT, "6%H}"));
+    view_typ s4(TYPED_LITERAL(CharT, "*<6hi"));
+    view_typ s5(TYPED_LITERAL(CharT, "*^4.4%ymm"));
+
+    vector<_Chrono_specs<CharT>> v0{_Chrono_specs<CharT>{'O', 'e'}};
+    test_parse_helper(parse_chrono_format_specs, s0, false, s0.size(), {.expected_chrono_specs = v0});
+
+    vector<_Chrono_specs<CharT>> v1{_Chrono_specs<CharT>{'l'}, _Chrono_specs<CharT>{'i'}, _Chrono_specs<CharT>{'t'}};
+    test_parse_helper(parse_chrono_format_specs, s1, false, s1.size(), {.expected_chrono_specs = v1});
+
+    vector<_Chrono_specs<CharT>> v2{
+        _Chrono_specs<CharT>{'\0', 'H'}, _Chrono_specs<CharT>{':'}, _Chrono_specs<CharT>{'\0', 'M'}};

```

Comment:
Suggestion: Extracting `_Chrono_specs<CharT>` to a short typedef would make this function much more readable.